.github/workflows/pythonapp-workflow.yml

```diff
@@ -0,0 +1,25 @@
+name: run-code-checks
+on: [push]
+jobs:
+  run-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.10
```

Comment:
I know you copied this from the app template, so it's not only a nit, it's a nit for myself...the name of this step should be `Set up Python` (w/o the 3.10), since `actions/setup-python@v5` will use `.python-version` to get the correct version

(I'll PR this change to the python template repo)